### PR TITLE
Add UI for Showing & Refreshing Rules

### PR DESCRIPTION
**Pull request by phillipbarron**
## What does this change?

Adds new UI in the rule-manager to Show and Refresh rules.

This will replace the existing UI in the Checker service. 

## How to test

Deploy, verify that all rules are available. The new UI uses an in-memory paginated view of the rules but also allows search. Search for text that appears in the Rules that you care about (we are searching across the pattern, id, description and type)

Try removing / adding a rule to the sheet and refreshing. Verify that the rule is added / removed as expected

## How can we measure success?

It works in the same way as the existing code in the checker UI

## Have we considered potential risks?

Think so?

## Images

Screenshot 2023-03-28 at 17 09 27

### CODE REVIEW

**Review by jonathonherbert — APPROVED** [review]
Looks lovely. A few minor comments.

**Comment by jonathonherbert** [comment]
Looks like we've a flaky test, post GHA migration, that's not solved by #109 – rerunning will hopefully solve.

File: apps/rule-manager/client/src/ts/components/hooks/rules-hook.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        setIsRefreshing(true);
+        try {
+            const updatedRulesResponse = await fetch(`${location}refresh`, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+            });
+            if (!updatedRulesResponse.ok) {
+                throw new Error('Failed to refresh rules');
```
[the remark this answers is not in the document]
Comment (phillipbarron) [reply apps/rule-manager/client/src/ts/components/hooks/rules-hook.ts:36]: Yeah, good shout

File: apps/rule-manager/client/src/ts/components/RulesTable.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    {
+        field: 'description',
+        name: 'Description'
+    }
+];
+
+const RulesTable = () => {
+    const {rules, isLoading, error, refreshRules, isRefreshing, setError} = useRules();
+
+    const [incremental, _] = useState(false);
```
[the remark this answers is not in the document]
Comment (phillipbarron) [reply apps/rule-manager/client/src/ts/components/RulesTable.tsx:74]: Generous of you to assume I have given this some thought 🤣  - I agree with true - cheers

**Pull request merged**